### DISCUSSION TITLE
N-Select change option properties

### DISCUSSION THREAD

**Discussion by AlexZCP** [tusen-ai/naive-ui#2786] [General]
By default  N-Select uses SelectOption  type as select options source.

SelectOption object mostly looks like that:

```
`          {
            label: 'Nowhere Man',
            value: 'song4'
          },
          {
            label: 'Think For Yourself',
            value: 'song5'
          },
          {
            label: 'The Word',
            value: 'song6'
          },
          {
            label: 'Michelle',
            value: 'song7',
            disabled: true
          }
`
```

But if I have a long data list of objects with another property than label and value.
For example I have similar: 

```
`          {
            name: 'Nowhere Man',
            id: 'song4'
          },
          {
            name: 'Think For Yourself',
            id: 'song5'
          },
```

How can i set this as options, not rewriting object itself. 
I assume need to use render-option property, but this make me change standart html structure of n-select option. 
Thats what I dont need,  I need just change name of properties with values and labels.

**Comment by 07akioni** [comment]
Currently I don't think map it is complex. I don't want add an API unless there are some scenarios without the API will be very hard to handle (maybe at least add 10 lines of code).

**Reply by 07akioni** [reply]
If you want us to add the API, you can explain that obstacles there would be without the API.
